rpc/namenode.go

```diff
@@ -109,19 +172,28 @@ func (c *NamenodeConnection) Execute(method string, req proto.Message, resp prot
 	defer c.reqLock.Unlock()
 
 	c.currentRequestID++
-	err := c.writeRequest(method, req)
+
+R:
+	err := c.resolveConnection()
 	if err != nil {
-		c.conn.Close()
 		return err
 	}
 
+	err = c.writeRequest(method, req)
+	if err != nil {
+		c.markFailure(err)
+		goto R
+	}
+
 	err = c.readResponse(method, resp)
 	if err != nil {
-		if _, ok := err.(*NamenodeError); !ok {
-			c.conn.Close() // TODO don't close on RPC failure
+		if nerr, ok := err.(*NamenodeError); ok {
+			if nerr.Exception != "org.apache.hadoop.ipc.StandbyException" {
+				return err
+			}
 		}
-
-		return err
+		c.markFailure(err)
```

Comment:
This is a bit confusing, but I think this treats `StandbyException`s the same as any non-`NamenodeError` error.

Reply:
Yep, the operation will only retry if it's either a non-`NamenodeError` or a `StandbyException`